### ISSUE TITLE
AndroidLintParser doesn't expose rule correctly

### ISSUE THREAD

**Issue by TWiStErRob** [tomasbjerre/violations-lib#39]
https://github.com/tomasbjerre/violations-lib/blob/master/src/main/java/se/bjurr/violations/lib/parsers/AndroidLintParser.java

uses
```
String id = getAttribute(issueChunk, "id");
String rule = getAttribute(issueChunk, "category");
...
.setRule(rule)
.setMessage(id + ": " + summary + "\n" + message + "\n" + explanation)
```
notice that the category is used for rule, which is wrong. The rule should be the ID as that's what's failed, category is like "Accessibility" which category can have 10 different rules inside it.

What can we do?

**Comment by tomasbjerre** [comment]
Ok.

So `.setRule(id)`.

Any suggestions on how to change `.setMessage(id + ": " + summary + "\n" + message + "\n" + explanation)` ? Or should the `category` not be included anywhere?

I never used AndroidLint myself so not really sure what is the best way of showing the findings.

**Comment by TWiStErRob** [comment]
I dropped you a mail with an example output of Android Lint, that should help you understand this.

re
>Any suggestions on how to change .setMessage(id + ": " + summary + "\n" + message + "\n" + explanation) ? Or should the category not be included anywhere?

I think your `Violation` model may be missing a category/group field, as PMD, Checkstyle, Lint, Detekt (the tools we use all have that). So I would say it's pretty common to have the rules/checks grouped by type.

**Comment by TWiStErRob** [comment]
Checkstyle: the groups on the left? http://checkstyle.sourceforge.net/checks.html
Hmm, it looks like it's only documentation it seems that the output doesn't support this (https://github.com/checkstyle/checkstyle/issues/5166).
I could open a PR that does a best effort to reproduce these categories? (i.e. a hash from check name to category)

Detekt: see example output https://github.com/arturbosch/detekt/raw/master/img/detekt_in_action.png and configuration https://github.com/arturbosch/detekt/blob/master/detekt-cli/src/main/resources/default-detekt-config.yml
I think re Detekt the problem is that it's read through Checkstyle's output format which doesn't have this field.

**Comment by tomasbjerre** [comment]
Ok. I'm pushing the change you suggested in your comment on my diff now.

You can give it a shot with a PR.

Regarding Detekt I think so as well. It is not in the checkstyle format. If you want to add a Detekt parser, that is great =)

**Comment by TWiStErRob** [comment]
Probably eventually, right now I'm working on integrating with your existing parsers. (using the lib to parse what we have and producing a nice output)

**Comment by TWiStErRob** [comment]
I think you can close this as you have the commit on master that fixes the id as rule.

**Comment by tomasbjerre** [comment]
I released 1.59 now with this change.
